store.js

```diff
@@ -1,15 +1,28 @@
-import { createStore } from 'redux';
+import { applyMiddleware, createStore } from 'redux';
+import thunkMiddleware from 'redux-thunk';
 
 const initialState = {
+  user: {
+    refreshingContacts: false,
+    contactItem: []
+  }
 };
 
 function reducer(state = initialState, action) {
   switch (action.type) {
+    case 'SET_USER_DATA':
+      return {
```

Comment:
Do we need `Object.assign({}, ...)` here?

Reply:
yes because state should be immutable

Reply:
Agreed, but I think the `Object.assign` is missing?

Reply:
you're right, sorry I thought you wanted to remove it. Added